pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -44,7 +197,91 @@ def resource_rep(cls) -> re.CompressedResourceOp:
 
 
 class ResourceCNOT(qml.CNOT, re.ResourceOperator):
-    """Resource class for CNOT"""
+    r"""Resource class for CNOT
+    
+    Resources:
+        There is no further decomposition provided for this gate.
+
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        raise re.ResourcesNotDefined
+
+    def resource_params(self) -> dict:
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceToffoli(qml.Toffoli, re.ResourceOperator):
+    """Resource class for Toffoli
+    
+    Resources:
+        The resources are obtained from the paper `Novel constructions for the fault-tolerant 
+        Toffoli gate <https://arxiv.org/pdf/1212.5069>`_. We summarize the cost as: 
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        raise re.ResourcesNotDefined
+
+    @staticmethod
+    def textbook_resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
```

Comment:
How should an alternative decomposition be called in our workflow?

Reply:
I imagine its something like: `re.ResourceToffoli.set_resources(re.ResourceToffoli.textbook_Resource_decomp)`